co_sim_io/impl/define.hpp

```diff
@@ -14,13 +14,16 @@
 #define CO_SIM_IO_DEFINE_INCLUDED
 
 // System includes
+#include <cstddef>
 #include <memory>
-#include <string>
-#include <unordered_map>
+#include <array>
 
 namespace CoSimIO {
 
-using SettingsType = std::unordered_map<std::string, std::string>;
+// signed integer type, 32 bit in 32 bit systems, but 64bit in 64 bit systems => like std::size_t but signed
+using IdType = std::ptrdiff_t;
```

Comment:
FYI @pooyan-dadvand, as discussed yesterday

having a typedef would allow us to change the type under the hood should it become necessary in the future